AppointmentReminders.Web/Controllers/AppointmentsController.cs

```diff
@@ -48,10 +94,34 @@ public ActionResult Edit(int? id)
                 return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
             }
 
-            // Find participant
+            var appointment = _repository.FindById(id.Value);
+            if (appointment == null)
+            {
+                return HttpNotFound();
+            }
 
-            return View(new Appointment());
+            ViewBag.Timezones = Timezones;
+            return View(appointment);
+        }
+
+        // POST: /Appointments/Edit/5
+        [HttpPost]
+        public ActionResult Edit([Bind(Include = "ID,Name,PhoneNumber,Time,Timezone")] Appointment appointment)
+        {
+            if (ModelState.IsValid)
+            {
+                _repository.Update(appointment);
+                return RedirectToAction("Details", new { id = appointment.Id });
+            }
+            return View(appointment);
+        }
 
+        // GET: Appointments/Delete/5
```

Comment:
we need to add a create an extension method, like a custom HtmlHelper, and create something like `DeleteLink`, and internally create a form for each link, the link should be then an input (submit).